lib/msf/core/auxiliary/http_crawler.rb

```diff
@@ -243,7 +243,8 @@ def crawl_target(t)
   # Specific module implementations should redefine this method
   # with whatever is meaningful to them.
   def crawler_process_page(t, page, cnt)
-    msg = "[#{"%.5d" % cnt}/#{"%.5d" % max_page_count}]    #{page.code || "ERR"} - #{@current_site.vhost} - #{page.url}"
+    return if page.nil? # Skip over pages that don't contain any info aka page is nil. We can't process these types of pages since there is no data to process.
```

Comment:
@gwillcox-r7 Any context on why this page object is nil in the first place? :eyes:

Without it context it looks like this might be fixing the side-effect, rather than the cause 👀

Reply:
I was assuming it was the way Anemone was using blocks as part of a custom iterator, likely causing a crash and nil return thanks to the implicit returns in Ruby, but I'm not nearly good enough at Ruby to discern what Anemone is weaving to be honest.

Reply:
You can get nils if there's a service accepting connection but returning nothing IIRC - this came up when scanning some video cameras which purport to speak HTTP but actually hold a non-HTTP service on 'em.

Reply:
@adfoster-r7 Is the explanation above sufficient or did you still have concerns before I landed this?

Reply:
Sounds good to me :+1: